src/secontext.c

```diff
@@ -161,44 +168,110 @@ selinux_getfilecon(struct tcb *tcp, const char *path, char **secontext,
 	if (!proc_pid)
 		return -1;
 
-	int rc = -1;
+	/*
+	 * The path may contain .../proc/self/... or .../proc/self<eos>.
+	 * We need to replace 'self' by the target PID, or else we would
+	 * resolve the path with 'self' being strace itself.
+	 * In some corner cases, it's impossible to guess so keep it as is if
+	 * we don't find any occurrence of /proc/self.
+	 * Finally there may be breakage if the pathname contains /proc/self
+	 * but has nothing to do with /proc.
+	 */
+
+	int rc;
 	char fname[PATH_MAX];
+	char buf[PATH_MAX + 1];
+	size_t l = strlcpy(fname, path, sizeof(fname));
+	if (l >= sizeof(fname) - 1)
+		return -1;
+	int proc_self_substituted = 0;
+
+	for (;;) {
+		char *s = strstr(fname, "/proc/self/");
```

Comment:
Sorry, I don't understand why do we need to substitute all occurrences of /proc/self/ in the string.

Reply:
Because you could have a query on /proc/self/root/proc/self/root for example. You cannot control how many /proc/self will be found, hence the loop.